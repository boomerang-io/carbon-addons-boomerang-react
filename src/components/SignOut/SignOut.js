import React from 'react';
import PropTypes from 'prop-types';
import { settings } from 'carbon-components';
import { Button } from 'carbon-components-react';
import {
  ModalHeader,
  ModalBody,
  ModalFooter,
} from 'carbon-components-react/lib/components/ComposedModal';

import HeaderMenuItem from '../HeaderMenuItem';

const { prefix } = settings;

SignOutContainer.propTypes = {
  signOutLink: PropTypes.string,
};

SignOutContainer.defaultProps = {};

function SignOutContainer({ signOutLink }) {
  return (
    <HeaderMenuItem
      text="Sign out"
      iconName="power"
      className={`${prefix}--bmrg-signout-container`}
    >
      {({ closeModal }) => {
        return (
          <>
            <ModalHeader title="Sign out?" closeModal={closeModal} />
            <ModalBody>
              <div className={`${prefix}--bmrg-signout`}>
                <p className={`${prefix}--bmrg-signout__message`}>
                  {"Are you sure you'd like to leave us?"}
                </p>
              </div>
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" onClick={closeModal}>
                Cancel
              </Button>
              <Button kind="primary" role="link" href={signOutLink}>
                Sign Out
              </Button>
            </ModalFooter>
          </>
        );
      }}
    </HeaderMenuItem>
  );
}

export default SignOutContainer;
