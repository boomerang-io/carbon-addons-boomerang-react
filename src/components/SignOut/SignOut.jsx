import React from "react";
import PropTypes from "prop-types";
import { prefix } from "../../internal/settings";
import { Button } from "@carbon/react";
import { ModalHeader, ModalBody, ModalFooter } from "@carbon/react";

import HeaderMenuItem from "../HeaderMenuItem";

SignOutContainer.propTypes = {
  signOutLink: PropTypes.string.isRequired,
};

SignOutContainer.defaultProps = {};

function SignOutContainer({ signOutLink }) {
  return (
    <HeaderMenuItem text="Sign out" iconName="power" className={`${prefix}--bmrg-signout-container`}>
      {({ closeModal }) => {
        return (
          <>
            <ModalHeader title="Sign out" closeModal={closeModal} />
            <ModalBody>
              <div className={`${prefix}--bmrg-signout`}>
                <p className={`${prefix}--bmrg-signout__message`}>{"Are you sure you'd like to leave us?"}</p>
              </div>
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" onClick={closeModal}>
                Cancel
              </Button>
              <Button kind="primary" role="link" href={signOutLink}>
                Sign Out
              </Button>
            </ModalFooter>
          </>
        );
      }}
    </HeaderMenuItem>
  );
}

export default SignOutContainer;
