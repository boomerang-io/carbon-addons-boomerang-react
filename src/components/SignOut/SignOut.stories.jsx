import Component from "./SignOut";

export default {
  title: "Platform/SignOut",
  component: Component,
};

export const SignOut = (args) => {
  return <Component {...args} />;
};
