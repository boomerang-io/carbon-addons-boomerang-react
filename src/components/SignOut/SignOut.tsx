import React from "react";
import { Button, ComposedModal, ModalHeader, ModalBody, ModalFooter } from "@carbon/react";
import HeaderMenuItem from "../Header/HeaderMenuItem";
import { Power } from "@carbon/react/icons";
import { prefix } from "../../internal/settings";

type Props = {
  closeModal: () => void;
  isOpen: boolean;
  signOutLink: string;
};

function SignOut({ closeModal, isOpen, signOutLink }: Props) {
  return (
    <ComposedModal
      open={isOpen}
      className={`${prefix}--bmrg-signout-container ${prefix}--bmrg-header-modal`}
      onClose={closeModal}
    >
      <ModalHeader title="Sign out" closeModal={closeModal} />
      <ModalBody>
        <div className={`${prefix}--bmrg-signout`}>
          <p className={`${prefix}--bmrg-signout__message`}>{"Are you sure you'd like to leave us?"}</p>
        </div>
      </ModalBody>
      <ModalFooter>
        <Button data-modal-primary-focus kind="secondary" onClick={closeModal}>
          Cancel
        </Button>
        <Button kind="primary" role="link" href={signOutLink}>
          Sign Out
        </Button>
      </ModalFooter>
    </ComposedModal>
  );
}

export default SignOut;

function SignOutMenuItem(props: Omit<Props, "isOpen" | "closeModal">) {
  const menuItemRef = React.useRef<HTMLButtonElement>(null);
  const [isOpen, setIsOpen] = React.useState(false);

  const handleClose = () => {
    setIsOpen(false);
    menuItemRef.current?.focus();
  };

  return (
    <>
      <HeaderMenuItem
        icon={<Power />}
        onClick={() => setIsOpen(!isOpen)}
        text="Sign Out"
        style={{ color: "red" }}
        type="button"
        variant="danger"
      />
      <SignOut isOpen={isOpen} closeModal={handleClose} {...props} />
    </>
  );
}

export { SignOutMenuItem };
