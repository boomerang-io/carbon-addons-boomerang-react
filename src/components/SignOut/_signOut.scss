@include exports('addons-bmrg-signOut') {
  .#{$prefix}--bmrg-signout-container {
    .#{$prefix}--modal-container {
      height: rem(226px);
      width: rem(500px);
    }

    .#{$prefix}--modal-content {
      margin-bottom: 0.5rem;
    }

    .#{$prefix}--bmrg-signout {
      margin-top: 1rem;
      margin-bottom: 2rem;
      &__message {
        color: #9fa5ad;
        font-size: rem(14px);
        line-height: rem(18px);
      }
    }

    .#{$prefix}--btn {
      &.#{$prefix}--btn--primary {
        min-width: 50%;
      }

      &.#{$prefix}--btn--secondary {
        max-width: 50%;
      }
    }
  }
}
