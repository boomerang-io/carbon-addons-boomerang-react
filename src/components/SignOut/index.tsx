export { default, SignOutMenuItem } from "./SignOut";
