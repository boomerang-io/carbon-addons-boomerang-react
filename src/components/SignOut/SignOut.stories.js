import React from 'react';
import { storiesOf } from '@storybook/react';
import SignOut from './SignOut';

storiesOf('SignOut', module).add('default', () => {
  return <SignOut />;
});
