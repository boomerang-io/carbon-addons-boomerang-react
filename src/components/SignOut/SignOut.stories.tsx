import React from "react";
import SignOut from "./SignOut";
import { headerModalProps } from "../../internal/helpers";

export default {
  title: "Platform/SignOut",
  component: SignOut,
  parameters: {
    info: {
      text: `
    Sign out component to handle users signing out of the platform
        `,
    },
  },
};

export const Default = (args) => {
  return <SignOut {...args} />;
};

Default.args = {
  signOutLink: "https://ibm.com",
  ...headerModalProps,
};
