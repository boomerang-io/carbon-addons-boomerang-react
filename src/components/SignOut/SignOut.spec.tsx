import React from "react";
import { expect, test} from "vitest";
import { render, screen } from "@testing-library/react";
import SignOut from "./SignOut";
import { headerModalProps } from "../../internal/helpers";

test("Signout - render correctly", () => {
  const { baseElement } = render(<div id="root"><SignOut signOutLink="https://ibm.com" {...headerModalProps} /></div>);
  expect(baseElement).toMatchSnapshot();
  expect(screen.getByText(/Are you sure you'd like to leave us\?/i)).toBeInTheDocument();
});
