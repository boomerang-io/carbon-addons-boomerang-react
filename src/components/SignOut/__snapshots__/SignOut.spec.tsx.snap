// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signout > snapshot 1`] = `
<body>
  <div>
    <div
      aria-hidden="false"
      class="cds--layer-one cds--modal is-visible cds--bmrg-signout-container cds--bmrg-header-modal"
      role="presentation"
    >
      <div
        aria-label="Sign out"
        aria-modal="true"
        class="cds--modal-container"
        role="dialog"
      >
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
        <div
          class="cds--modal-container-body"
        >
          
          <div
            class="cds--modal-header"
          >
            <h2
              class="cds--modal-header__heading cds--type-beta"
            >
              Sign out
            </h2>
            <div
              class="cds--modal-close-button"
            >
              <span
                class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--left cds--tooltip cds--icon-tooltip"
              >
                <div
                  class="cds--tooltip-trigger__wrapper"
                >
                  <button
                    aria-label="Close"
                    aria-labelledby="tooltip-2"
                    class="cds--btn--icon-only cds--modal-close cds--btn cds--btn--primary"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="cds--modal-close__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                      />
                    </svg>
                  </button>
                </div>
                <span
                  aria-hidden="true"
                  class="cds--popover"
                  id="tooltip-2"
                  role="tooltip"
                >
                  <span
                    class="cds--popover-content cds--tooltip-content"
                  >
                    Close
                  </span>
                  <span
                    class="cds--popover-caret"
                  />
                </span>
              </span>
            </div>
          </div>
          <div
            class="cds--layer-two cds--modal-content"
          >
            <div
              class="cds--bmrg-signout"
            >
              <p
                class="cds--bmrg-signout__message"
              >
                Are you sure you'd like to leave us?
              </p>
            </div>
          </div>
          <div
            aria-busy="false"
            class="cds--modal-footer cds--btn-set"
          >
            <button
              class="cds--btn cds--btn--secondary"
              data-modal-primary-focus="true"
              type="button"
            >
              Cancel
            </button>
            <a
              class="cds--btn cds--btn--primary"
              href="https://useboomerang.io"
              role="link"
            >
              Sign Out
            </a>
          </div>
        </div>
        <button
          class="cds--visually-hidden"
          type="button"
        >
          Focus sentinel
        </button>
      </div>
    </div>
  </div>
</body>
`;
